### PR TITLE
Refactored Partner Registration to registration and validation

### PR DESCRIPTION
**Pull request by danbar0kal**
Added a new action to only validate the new partner, without registering it

### CODE REVIEW

File: alpha/apps/kaltura/lib/myPartnerRegistration.class.php [hunk 1]
```diff
@@ -419,6 +431,18 @@ public function initNewPartner($partner_name , $contact, $email, $ID_is_for, $SD
 
 			return array($newPartner->getId(), $newSubPartnerId, $newAdminKuserPassword, $newPassHashKey);
 		}
+		catch (SignupException $se)
+		{
+			if (strpos($se->getMessage(),'Invalid'))
```
Comment (MosheMaorKaltura) [inline alpha/apps/kaltura/lib/myPartnerRegistration.class.php:436]: Validate the exception type by the code and not by the message

File: alpha/apps/kaltura/lib/myPartnerRegistration.class.php [hunk 2]
```diff
@@ -419,6 +431,18 @@ public function initNewPartner($partner_name , $contact, $email, $ID_is_for, $SD
...
+			{
+				$this->addMarketoCampaignId($existingPartner, myPartnerUtils::MARKETO_WRONG_PASSWORD, $partner);
+			}
+			else
+			{
```
Comment (MosheMaorKaltura) [inline alpha/apps/kaltura/lib/myPartnerRegistration.class.php:441]: Validate both assuming the new type can be added in the future

File: api_v3/services/PartnerService.php [hunk 3]
```diff
@@ -16,79 +16,118 @@ protected function partnerRequired($actionName)
...
 	 *
 	 * @throws APIErrors::PARTNER_REGISTRATION_ERROR
 	 */
-	public function registerAction( KalturaPartner $partner , $cmsPassword = "" , $templatePartnerId = null, $silent = false)
+	public function registrationValidationAction(KalturaPartner $partner, $cmsPassword = "" ,$templatePartnerId = null, $silent = false)
 	{
 		KalturaResponseCacher::disableCache();
 		
-		$dbPartner = $partner->toPartner();
+		$tmpPartner = clone $partner;
```
Comment (MosheMaorKaltura) [inline api_v3/services/PartnerService.php:38]: why is tmpPartner needed here? you can use $partner for you actions.

File: api_v3/services/PartnerService.php [hunk 4]
```diff
@@ -16,79 +16,118 @@ protected function partnerRequired($actionName)
...
+	{
+		KalturaResponseCacher::disableCache();
+		
+		try
+		{
+			$c = new Criteria();
+			$c->addAnd(UserLoginDataPeer::LOGIN_EMAIL, $partner->adminEmail, Criteria::EQUAL);
+			$existingUser = UserLoginDataPeer::doSelectOne($c);
+			/* @var $exisitingUser UserLoginData */
+			$this->registrationValidationAction($partner, $cmsPassword, $templatePartnerId, $silent);
```
Comment (MosheMaorKaltura) [inline api_v3/services/PartnerService.php:119]: Assumption is that the action is being called from the controller and we should avoid internal calling +  you can call $partner_registration->validateNewPartner here, all the inputs arguments are here anyway

**Comment by kaltura-hooks** [comment]
Hi @danbar0kal,
Thank you for contributing this pull request!
Please sign the Kaltura CLA so we can review and merge your contribution.
Learn more at http://bit.ly/KalturaContrib

**Pull request merged**